packages/daemon/src/mail.js

```diff
@@ -58,11 +60,32 @@ export const makeMailboxMaker = ({
       );
     };
 
-    const terminate = async petName => {
-      const formulaIdentifier = lookupFormulaIdentifierForName(petName);
-      if (formulaIdentifier === undefined) {
-        throw new TypeError(`Unknown pet name: ${q(petName)}`);
+    /**
+     * @param {string | string[]} petNameOrPath - A pet name or a sequence of pet names.
+     * @returns {Promise<void>} A promise that resolves when the name is terminated.
+     */
+    const terminate = async petNameOrPath => {
+      const petNamePath = petNamePathFrom(petNameOrPath);
+      console.log(`trace:terminate1: ${petNamePath.join(', ')}`);
+
+      let formulaIdentifier;
+      if (petNamePath.length === 1) {
+        formulaIdentifier = lookupFormulaIdentifierForName(petNamePath[0]);
+        if (formulaIdentifier === undefined) {
+          throw new TypeError(`Unknown pet name: ${q(petNamePath[0])}`);
+        }
+      } else {
+        // eslint-disable-next-line no-use-before-define
+        const { value } = await provideLookupFormula(petNamePath);
+        // eslint-disable-next-line no-use-before-define
+        [formulaIdentifier] = await reverseLookup(value);
+        if (formulaIdentifier === undefined) {
+          throw new TypeError(
+            `Value at ${q(petNamePath.join(', '))} cannot be terminated.`,
+          );
+        }
```

Comment:
Can `provideLookupFormula` return the `formulaIdentifier`? There are other functions like that.

Reply:
This is already the case.